File: backend/app/models/agent.py
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any
from datetime import datetime


class AgentCreate(BaseModel):
    """Model for creating a new AI agent"""
    name: str = Field(..., min_length=1, max_length=100)
    category: str = Field(..., description="inquiry|booking|order_status|support|follow_up|sales|general")
    language: str = Field(default="en", description="en|ta")
    system_prompt: str = Field(..., min_length=10)
    first_message: str = Field(..., min_length=5)
    description: Optional[str] = None


class AgentResponse(BaseModel):
    """Response model for an agent"""
    id: str                          # Supabase row id
    vapi_assistant_id: str           # Vapi assistant id
    name: str
    category: str
    language: str
    system_prompt: str
    first_message: str
    description: Optional[str] = None
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/agent.ts
export type AgentCategory = 'inquiry' | 'booking' | 'order_status' | 'support' | 'follow_up' | 'sales' | 'general'
export type AgentLanguage = 'en' | 'hi' | 'ta' | 'te' | 'kn' | 'ml' | 'bn' | 'mr' | 'gu' | 'es' | 'fr' | 'ar'

export interface Agent {
  id: string
  vapi_assistant_id: string
  name: string
  category: AgentCategory
  language: AgentLanguage
  system_prompt: string
  first_message: string
  description?: string
  created_at: string
  updated_at: string
}

export interface AgentVoice {
  provider: string
  voiceId: string
}

export interface AgentTranscriber {
  provider: string
  language: string
  model?: string
}

export interface AgentModel {
  provider: string
  model: string
}

export interface AgentCreate {
  name: string
  category: AgentCategory
  language: AgentLanguage
  system_prompt: string
  first_message: string
  description?: string
  /** User-selected voice — if omitted the API falls back to the language default */
  voice?: AgentVoice
  /** User-selected transcriber — if omitted the API falls back to the language default */
  transcriber?: AgentTranscriber
  /** User-selected LLM — if omitted the API falls back to gemini-2.0-flash-lite */
  llm?: AgentModel
}
